src/App.tsx

```diff
@@ -1,14 +1,52 @@
-import React from 'react';
+import { useCallback, useMemo, useState } from 'react';
+import { debounce } from 'lodash';
 import './App.scss';
+import { Autocomplete } from './components/Autocomplete';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  delay: number;
+};
+
+export const App: React.FC<Props> = ({ delay = 1000 }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [showList, setShowList] = useState(true);
+  const [onSelected, setOnSelected] = useState('No selected person');
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const getVisiblePeople = () => {
+    return peopleFromServer
+      .filter(person => person.name.toLocaleLowerCase()
+        .includes(appliedQuery.toLocaleLowerCase()));
+  };
+
+  const visiplePeople = useMemo(
+    getVisiblePeople,
+    [peopleFromServer, appliedQuery],
+  );
+
+  const handlePersonSelect = (person: Person) => {
+    setOnSelected(`${person.name} (${person.born} = ${person.died})`);
+    setShowList(false);
+    setQuery(person.name);
+  };
+
+  const handleInputChange = (inputValue: string) => {
```

Comment:
The `handleInputChange` function should be moved to the `Autocomplete` component.